sql/src/main/java/io/crate/metadata/sys/SysSnapshotsTableInfo.java

```diff
@@ -101,11 +102,26 @@ public Routing getRouting(WhereClause whereClause, @Nullable String preference)
         Map<String, Map<String, List<Integer>>> locations = new TreeMap<>();
         Map<String, List<Integer>> tableLocation = new TreeMap<>();
         tableLocation.put(IDENT.fqn(), null);
-        // route to master node, because RepositoriesService (and so snapshots info) is only available there
-        locations.put(clusterService.state().nodes().masterNodeId(), tableLocation);
+        // route to random master or data node,
+        // because RepositoriesService (and so snapshots info) is only available there
+        locations.put(randomMasterOrDataNode().id(), tableLocation);
         return new Routing(locations);
     }
 
+    private DiscoveryNode randomMasterOrDataNode() {
+        Random random = new Random();
+        int randomIdx = random.nextInt(clusterService.state().nodes().masterAndDataNodes().size());
+        Iterator<DiscoveryNode> it = clusterService.state().nodes().masterAndDataNodes().valuesIt();
+        int currIdx = 0;
+        while (it.hasNext()) {
+            if (currIdx == randomIdx) {
+                return it.next();
+            }
+            currIdx++;
+        }
+        return null;
```

Comment:
wouldn't it make sense to raise an AssertionError here? this case shouldn't happen